components/kyma-environment-broker/internal/orchestration/kyma/manager.go

```diff
@@ -0,0 +1,140 @@
+package kyma
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/orchestration"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type upgradeKymaManager struct {
+	db                  storage.Orchestrations
+	resolver            orchestration.RuntimeResolver
+	kymaUpgradeExecutor process.Executor
+	log                 logrus.FieldLogger
+}
+
+func NewUpgradeKymaManager(db storage.Orchestrations, kymaUpgradeExecutor process.Executor, resolver orchestration.RuntimeResolver, log logrus.FieldLogger) process.Executor {
+	return &upgradeKymaManager{
+		db:                  db,
+		resolver:            resolver,
+		kymaUpgradeExecutor: kymaUpgradeExecutor,
+		log:                 log,
+	}
+}
+
+// Execute reconciles runtimes for a given orchestration
+func (u *upgradeKymaManager) Execute(orchestrationID string) (time.Duration, error) {
+	o, err := u.db.GetByID(orchestrationID)
+	if err != nil {
+		return 0, errors.Wrapf(err, "while getting orchestration %s", orchestrationID)
+	}
+
+	dto := orchestration.Parameters{}
+	if o.Parameters.Valid {
+		err = json.Unmarshal([]byte(o.Parameters.String), &dto)
+		if err != nil {
+			return 0, errors.Wrap(err, "while unmarshalling parameters")
+		}
+	}
+
+	targets := dto.Targets
+	if targets.Include == nil || len(targets.Include) == 0 {
+		targets.Include = []internal.RuntimeTarget{{Target: internal.TargetAll}}
+	}
+
+	operations, err := u.resolveOperations(o, dto)
+	if err != nil {
+		return 0, errors.Wrap(err, "while resolving operations")
+	}
+
+	state := internal.Succeeded
+	desc := fmt.Sprintf("scheduled %d operations", len(operations))
+
+	if err = u.updateOrchestration(o, state, desc, operations); err != nil {
+		return 0, errors.Wrap(err, "while updating orchestration")
+	}
+	if len(operations) == 0 {
+		return 0, nil
+	}
+
+	// TODO(upgrade): support many strategies
+	strategy := orchestration.NewInstantOrchestrationStrategy(u.kymaUpgradeExecutor, u.log)
+	_, err = strategy.Execute(operations, dto.Strategy)
+	if err != nil {
+		return 0, errors.Wrap(err, "while executing instant upgrade strategy")
+	}
+
+	// TODO(upgrade): check UpgradeKymaOperations in the loop to assert orchestration state
+	result, err := u.checkOperationsResults(operations)
+	if err != nil {
+		return 0, errors.Wrap(err, "while checking operations results")
+	}
+	if !result {
+		state = internal.Failed
+	}
+	err = u.updateOrchestration(o, state, desc, operations)
+	if err != nil {
+		return 0, errors.Wrap(err, "while updating orchestration")
```

Comment:
you can use the functionality of repeating operations

Reply:
done, in case of failed upgrade I repeat the processing after 1 min